spec/counter_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+RSpec.describe Kitchen::Counter do
+  describe '.initialize' do
+    it 'creates a counter' do
+      expect(Kitchen::Counter.new).to be_truthy
+    end
+  end
+  describe '.increment' do
+    counter = Kitchen::Counter.new
+    it 'increments by 1 given no args' do
+      val = counter.instance_variable_get(:@value)
+      counter.increment
+      expect(counter.instance_variable_get(:@value)).to eq(val + 1)
+    end
+    it 'increments by 2' do
+      val = counter.instance_variable_get(:@value)
+      counter.increment(by: 2)
+      expect(counter.instance_variable_get(:@value)).to eq(val + 2)
+    end
+    it 'increments by -5' do
+      val = counter.instance_variable_get(:@value)
+      counter.increment(by: -5)
+      expect(counter.instance_variable_get(:@value)).to eq(val - 5)
+    end
+  end
+  describe '.get' do
+    it 'gets the counter value' do
+      counter = Kitchen::Counter.new
+      expect(counter.get).to eq(counter.instance_variable_get(:@value))
```

Comment:
I think it is ok to only test the public interface and not look at the internal value.  We know that `get` just returns `@value`, and our other code will make sure that `get` gives good results.

Reply:
So to clarify, you mean I should just use `get` in my other specs instead of `instance_variable_get`, and take out this spec?

Reply:
Hah, just scrolled down and saw your other comment. Disregard!